In [ ]:
# Célula 1: Instalar a biblioteca do Kaggle
!pip install kagglehub

In [ ]:
import kagglehub
import pandas as pd

# O código vai baixar os arquivos e salvar o caminho na variável 'path'
path = kagglehub.dataset_download("nawazkhan3251/student-performance-and-behavioral-analytics-dataset")

# Como sabemos que o nome do arquivo é esse, já montamos o caminho completo
caminho_csv = f"{path}/hybrid_student_performance_1200.csv"

# Carregando no Pandas!
df = pd.read_csv(caminho_csv)

Using Colab cache for faster access to the 'student-performance-and-behavioral-analytics-dataset' dataset.


In [ ]:
df.head()

,student_id,timestamp,year_class,program_stream,age,gender,cgpa_category,academic_satisfaction,study_hours_daily,daily_productivity,...,projects_internships,programming_foundation,events_participation,assignments_on_time,attendance_percentage,strongest_asset,internal_barrier,external_resources,external_pressure,performance_risk_level
0,STU0001,3/29/2026 20:54:43,Second Year (SY),BCA,20,Female,8.5 – 9.4,Neutral,Less than 1 hour,2.0,...,"Yes, actively working on projects/internship","Basic knowledge, learning while practicing",Occasionally participate in events,Often,Above 85%,"Technical/Hard Skills (Coding, Math, Logic)",Lack of Consistency or Determination (Difficul...,Occasionally (When needed),No Impact (Fully supportive environment),Moderate Risk
1,STU0002,3/29/2026 21:20:15,Second Year (SY),BCA,20,Male,8.5 – 9.4,Unsatisfied,1–2 hours,1.0,...,"Yes, actively working on projects/internship","Limited knowledge, theoretical only","Rarely participate, mostly observe",Often,76% – 85%,"Creative/Design Skills (Innovation, UI/UX, Con...",Lack of Consistency or Determination (Difficul...,Occasionally (When needed),Moderate Impact (Occasional disruption),Moderate Risk
2,STU0003,3/29/2026 21:21:35,Second Year (SY),BCA,19,Male,5.0 – 6.9,Neutral,1–2 hours,2.0,...,"Yes, actively working on projects/internship","Basic knowledge, learning while practicing","Rarely participate, mostly observe",Often,Less than 50%,"Technical/Hard Skills (Coding, Math, Logic)",Difficulty with Focus / Concentration,Rarely (Passive),High Impact (Frequent disruption),High Risk
3,STU0004,3/29/2026 21:27:44,Second Year (SY),BCA,19,Female,8.5 – 9.4,Satisfied,1–2 hours,3.0,...,Planning to start a project/internship soon,"Basic knowledge, learning while practicing","Rarely participate, mostly observe",Often,Above 85%,"Management/Execution (Planning, Organizing, Di...",Procrastination / Low Motivation,Occasionally (When needed),High Impact (Frequent disruption),Low Risk
4,STU0005,3/29/2026 22:05:14,Second Year (SY),BCA,20,Female,8.5 – 9.4,Neutral,1–2 hours,2.0,...,Planning to start a project/internship soon,"Basic knowledge, learning while practicing",Occasionally participate in events,Often,76% – 85%,"Management/Execution (Planning, Organizing, Di...",Lack of Consistency or Determination (Difficul...,Occasionally (When needed),High Impact (Frequent disruption),High Risk


In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1200 entries, 0 to 1199
Data columns (total 36 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   student_id              1200 non-null   object 
 1   timestamp               1200 non-null   object 
 2   year_class              1200 non-null   object 
 3   program_stream          1200 non-null   object 
 4   age                     1200 non-null   int64  
 5   gender                  1200 non-null   object 
 6   cgpa_category           1200 non-null   object 
 7   academic_satisfaction   1200 non-null   object 
 8   study_hours_daily       1200 non-null   object 
 9   daily_productivity      1166 non-null   float64
 10  revision_frequency      1166 non-null   object 
 11  focus_duration          1157 non-null   object 
 12  screen_time_non_study   1168 non-null   object 
 13  main_distractor         1200 non-null   object 
 14  study_consistency       1200 non-null   

In [ ]:
df['attendance_percentage'].value_counts()

,count
attendance_percentage,
76% – 85%,298
66% – 75%,294
50% – 65%,248
Above 85%,225
Less than 50%,135


In [ ]:
df.isnull().sum()

,0
student_id,0
timestamp,0
year_class,0
program_stream,0
age,0
gender,0
cgpa_category,0
academic_satisfaction,0
study_hours_daily,0
daily_productivity,34


In [ ]:
df[df.isnull().any(axis=1)]

,student_id,timestamp,year_class,program_stream,age,gender,cgpa_category,academic_satisfaction,study_hours_daily,daily_productivity,...,projects_internships,programming_foundation,events_participation,assignments_on_time,attendance_percentage,strongest_asset,internal_barrier,external_resources,external_pressure,performance_risk_level
15,STU0016,03/18/2026 13:40:00,First Year (FY),BCA,17,Female,9.5 – 10.0,Neutral,1–2 hours,1.0,...,Planning to start a project/internship soon,Strong foundation in core concepts,Occasionally participate in events,Rarely,50% – 65%,"Technical/Hard Skills (Coding, Math, Logic)",Difficulty with Focus / Concentration,Occasionally (When needed),High Impact (Frequent disruption),Moderate Risk
16,STU0017,04/01/2026 10:44:00,Second Year (SY),BCA,20,Male,8.5 – 9.4,Very satisfied,More than 2 hours,4.0,...,"Not currently, but intend to in the future",Strong foundation in core concepts,"Rarely participate, mostly observe",Sometimes,Above 85%,"Creative/Design Skills (Innovation, UI/UX, Con...",Poor Time Management / Over-scheduling,Never (Unaware or Not interested),Moderate Impact (Occasional disruption),Low Risk
19,STU0020,03/27/2026 14:29:00,Third Year (TY),BCA,21,Female,7.0 – 8.4,Satisfied,Less than 1 hour,3.0,...,"Not currently, but intend to in the future","Limited knowledge, theoretical only","Rarely participate, mostly observe",Sometimes,Less than 50%,"Management/Execution (Planning, Organizing, Di...",Lack of Consistency or Determination (Difficul...,Rarely (Passive),Low Impact (Rarely affects study),High Risk
22,STU0023,04/01/2026 23:56:00,First Year (FY),BCA,18,Female,8.5 – 9.4,Neutral,More than 2 hours,3.0,...,"Yes, actively working on projects/internship","Basic knowledge, learning while practicing","Rarely participate, mostly observe",Often,76% – 85%,"Technical/Hard Skills (Coding, Math, Logic)",Procrastination / Low Motivation,Never (Unaware or Not interested),Low Impact (Rarely affects study),Moderate Risk
23,STU0024,04/03/2026 08:32:00,Third Year (TY),BSc IT,20,Male,8.5 – 9.4,Very satisfied,Less than 1 hour,2.0,...,"Yes, actively working on projects/internship","Basic knowledge, learning while practicing","Rarely participate, mostly observe",Always,76% – 85%,"Technical/Hard Skills (Coding, Math, Logic)",Lack of Consistency or Determination (Difficul...,Rarely (Passive),Moderate Impact (Occasional disruption),Moderate Risk
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1171,STU1172,04/11/2026 22:23:00,First Year (FY),BSc IT,17,Male,8.5 – 9.4,Satisfied,1–2 hours,NaN,...,Planning to start a project/internship soon,NaN,Never participate in such events,Rarely,Above 85%,"Soft Skills (Communication, Leadership, Teamwork)",Poor Time Management / Over-scheduling,Occasionally (When needed),High Impact (Frequent disruption),Low Risk
1173,STU1174,04/09/2026 16:47:00,Final Year,BSc IT,21,Female,9.5 – 10.0,Satisfied,1–2 hours,4.0,...,Planning to start a project/internship soon,"Basic knowledge, learning while practicing",Never participate in such events,Always,50% – 65%,"Management/Execution (Planning, Organizing, Di...",Procrastination / Low Motivation,NaN,High Impact (Frequent disruption),Low Risk
1192,STU1193,04/14/2026 12:29:00,Final Year,BCA,21,Male,5.0 – 6.9,Very unsatisfied,Less than 1 hour,1.0,...,Planning to start a project/internship soon,"Basic knowledge, learning while practicing",Never participate in such events,Often,66% – 75%,"Creative/Design Skills (Innovation, UI/UX, Con...",Difficulty with Focus / Concentration,Rarely (Passive),High Impact (Frequent disruption),High Risk
1196,STU1197,03/19/2026 07:38:00,First Year (FY),BCA,18,Male,9.5 – 10.0,Unsatisfied,1–2 hours,3.0,...,Planning to start a project/internship soon,Strong foundation in core concepts,Occasionally participate in events,Sometimes,76% – 85%,"Technical/Hard Skills (Coding, Math, Logic)",Lack of Consistency or Determination (Difficul...,Rarely (Passive),Moderate Impact (Occasional disruption)

In [ ]:
colunas_texto = ['revision_frequency',

'focus_duration',

'screen_time_non_study',

'sleep_hours',

'online_courses',

'programming_foundation',

'events_participation',

'external_resources',

'external_pressure']

for colunas in colunas_texto:
  df[colunas] = df[colunas].fillna("Não informado")

colunas_numericas = ['daily_productivity',

'energy_level',

'stress_level',

'routine_rating']

for coluna in colunas_numericas:
  df[coluna] = df[coluna].fillna(df[coluna].median())

print(df.isnull().sum())

student_id                0
timestamp                 0
year_class                0
program_stream            0
age                       0
gender                    0
cgpa_category             0
academic_satisfaction     0
study_hours_daily         0
daily_productivity        0
revision_frequency        0
focus_duration            0
screen_time_non_study     0
main_distractor           0
study_consistency         0
tasks_on_time             0
preparation_status        0
career_goal_clarity       0
skills_developing         0
energy_level              0
stress_level              0
routine_rating            0
sleepy_during_study       0
sleep_hours               0
career_interest           0
online_courses            0
projects_internships      0
programming_foundation    0
events_participation      0
assignments_on_time       0
attendance_percentage     0
strongest_asset           0
internal_barrier          0
external_resources        0
external_pressure         0
performance_risk_lev

📊 Análise Exploratória e Criação de Gráficos


In [ ]:
analise_1 = df.groupby('study_hours_daily')['cgpa_category'].value_counts()
print(analise_1)

study_hours_daily  cgpa_category
1–2 hours          7.0 – 8.4        189
                   8.5 – 9.4        177
                   5.0 – 6.9         99
                   9.5 – 10.0        91
Less than 1 hour   7.0 – 8.4        196
                   5.0 – 6.9        118
                   8.5 – 9.4         85
                   9.5 – 10.0        33
More than 2 hours  8.5 – 9.4        100
                   9.5 – 10.0        58
                   7.0 – 8.4         35
                   5.0 – 6.9         19
Name: count, dtype: int64


In [ ]:
import plotly.express as px
df_grafico1 = df.groupby('study_hours_daily')['cgpa_category'].value_counts().reset_index(name='Quantity')
display(df_grafico1)

,study_hours_daily,cgpa_category,Quantity
0,1–2 hours,7.0 – 8.4,189
1,1–2 hours,8.5 – 9.4,177
2,1–2 hours,5.0 – 6.9,99
3,1–2 hours,9.5 – 10.0,91
4,Less than 1 hour,7.0 – 8.4,196
5,Less than 1 hour,5.0 – 6.9,118
6,Less than 1 hour,8.5 – 9.4,85
7,Less than 1 hour,9.5 – 10.0,33
8,More than 2 hours,8.5 – 9.4,100
9,More than 2 hours,9.5 – 10.0,58


In [ ]:
grafico1 = px.bar(
    df_grafico1,
    x='study_hours_daily',
    y='Quantity',
    color='cgpa_category',
    title='Impacto das Horas de Estudo na Nota Final (CGPA)',
    barmode='group',
    text_auto = 'true',
)

grafico1.show()

In [ ]:
analise_1_pct = (
    df.groupby('study_hours_daily')['cgpa_category']
      .value_counts(normalize=True)
)

print(analise_1_pct)

study_hours_daily  cgpa_category
1–2 hours          7.0 – 8.4        0.339928
                   8.5 – 9.4        0.318345
                   5.0 – 6.9        0.178058
                   9.5 – 10.0       0.163669
Less than 1 hour   7.0 – 8.4        0.453704
                   5.0 – 6.9        0.273148
                   8.5 – 9.4        0.196759
                   9.5 – 10.0       0.076389
More than 2 hours  8.5 – 9.4        0.471698
                   9.5 – 10.0       0.273585
                   7.0 – 8.4        0.165094
                   5.0 – 6.9        0.089623
Name: proportion, dtype: float64


In [ ]:
df_grafico1_pct = (
    df.groupby('study_hours_daily')['cgpa_category']
      .value_counts(normalize=True)
      .reset_index(name='Percentual')
)

In [ ]:
grafico1_pct = px.bar(
    df_grafico1_pct,
    x='study_hours_daily',
    y='Percentual',
    color='cgpa_category',
    title='Distribuição percentual do CGPA por horas de estudo',
    barmode='group',
    text_auto='.1%'
)

grafico1_pct.show()

### Análise de Desempenho vs. Horas de Estudo

Ao cruzar a carga horária diária de estudos com as notas (CGPA) dos alunos, identifiquei uma correlação direta e muito clara entre o esforço e o resultado final. O gráfico de valores absolutos nos deu uma visão geral do volume de alunos, mas foi a visão percentual (normalizada) que revelou o real impacto da dedicação:

* **O Grupo de Elite:** Alunos que estudam mais de 2 horas diárias dominam o topo acadêmico. Quase metade desse grupo (47.2%) atinge notas entre 8.5 e 9.4, e eles são os recordistas na faixa de excelência máxima (9.5 - 10.0), com mais de 27% dos alunos concentrados ali.
* **O Risco da Baixa Carga Horária:** No extremo oposto, a base de alunos que estuda menos de 1 hora por dia apresenta os piores resultados. A maioria fica presa na faixa intermediária (7.0 - 8.4), e este grupo possui a maior taxa de alunos com desempenho muito abaixo (5.0 - 6.9), ultrapassando os 27%.

**Conclusão:** O tempo de estudo é um fator determinante e previsível para o sucesso ou fracasso acadêmico neste cenário.

In [ ]:
grafico2_pct = df.groupby("projects_internships")["cgpa_category"].value_counts(normalize=True).reset_index(name="Percentual")

df_grafico2 = px.bar(
    grafico2_pct,
    x="projects_internships",
    y="Percentual",
    color="cgpa_category",
    title='O peso dos Projetos e Estágios nas Notas (CGPA)',
    barmode='group',
    text_auto='.1%'
)

df_grafico2.show()

### O Peso dos Projetos e Estágios no Desempenho

Ao cruzar o engajamento prático dos alunos (Projetos/Estágios) com suas notas, os dados quebram o mito de que trabalhar e estudar simultaneamente afunda o boletim:

* **O Topo é Prático:** O grupo de alunos que já atua ativamente no mercado ou em projetos ("Yes, actively working...") possui a maior concentração de notas máximas (9.5 - 10.0), atingindo 17.6%.
* **O Risco da Inércia:** Por outro lado, os alunos que não fazem estágio e nem planejam fazer no momento ("Not currently...") formam o grupo com a maior taxa de alunos com notas baixas na faixa de 5.0 - 6.9 (atingindo 21.1%).
* **O Equilíbrio do Planejamento:** Os alunos que estão organizando a vida para começar um estágio em breve ("Planning to start...") apresentam um cenário de muito equilíbrio e a menor taxa de notas baixas de toda a turma (17.9%).

**Conclusão:** Ter a "mão na massa" (ou estar se planejando ativamente para isso) puxa o desempenho do aluno para cima. A experiência prática atua como um motor para o estudo teórico, e não como uma âncora.

In [ ]:
grafico3_pct = df.groupby("stress_level")["cgpa_category"].value_counts(normalize=True).reset_index(name="Percentual")

df_grafico3 = px.bar(
    grafico3_pct,
    x="stress_level",
    y="Percentual",
    color="cgpa_category",
    title="Desempenho por nível de Estresse",
    barmode="group",
    text_auto='.1%'
)

df_grafico3.show()

### O Paradoxo do Estresse no Desempenho Acadêmico

Ao investigar a relação entre o Nível de Estresse (1 a 5) e as notas (CGPA), os dados revelaram um cenário não linear, quebrando a expectativa comum de que o estresse elevado sempre destrói o desempenho acadêmico:

* **Os Mais Estressados:** Surpreendentemente, os alunos com o nível máximo de estresse (Nível 5) possuem a maior taxa de notas de excelência da turma (19.1% na faixa de 9.5 - 10.0). Isso indica que a alta performance, muitas vezes, cobra um preço alto na saúde mental do estudante, ou que a própria cobrança por perfeição gera esse estresse.
* **O "Vale" do Nível Médio:** O pior cenário acadêmico não está nos extremos, mas no meio. Alunos com Nível 3 de estresse possuem a menor taxa de notas máximas (12.5%) e o maior pico de notas críticas (23.1% na faixa de 5.0 - 6.9).
* **O Equilíbrio Ideal:** O Nível 1 (baixo estresse) também apresenta ótimos resultados, com 18.5% de alunos na faixa de excelência, provando que é possível chegar ao topo sem sacrificar a saúde.

**Conclusão:** O estresse extremo não impede notas altas (pelo contrário, eles caminham juntos no topo), mas a falta de gerenciamento emocional ou a desorganização que gera um estresse "intermediário" parece ser fatal para o boletim.

In [ ]:
grafico4_pct = df.groupby("sleep_hours")["cgpa_category"].value_counts(normalize=True).reset_index(name="Percentual")

df_grafico4 = px.bar(
    grafico4_pct,
    x="sleep_hours",
    y="Percentual",
    title="Relação de Desempenho e Sono",
    color="cgpa_category",
    barmode="group",
    text_auto='.1%'
)

df_grafico4.show()

### O Ponto de Equilíbrio do Sono

Ao analisar a relação entre as horas de sono e as notas finais (CGPA), descobrimos que a regra de "quanto mais, melhor" não se aplica ao descanso acadêmico. O gráfico revela uma curva de eficiência muito clara:

* **A Estratégia Ideal:** O pico de excelência acadêmica não está nos extremos. Alunos que dormem entre 6 e 7 horas possuem a maior concentração de notas máximas (17.0% na faixa de 9.5 - 10.0). É o balanço perfeito entre tempo de estudo e recuperação mental.
* **O Esforço Noturno:** Alunos que dormem pouco (4-5 horas) superam aqueles que dormem muito. Eles possuem 14.5% de alunos na faixa de excelência. Isso sugere que a privação de sono, embora não seja o cenário ideal, geralmente está associada a horas extras de estudo e dedicação (o que cruza com nossos dados de alto estresse e altas notas).
* **A Inércia do Sono Excessivo:** Dormir mais de 8 horas apresenta a menor conversão para notas de elite (apenas 12.5%), indicando que o excesso de descanso pode estar associado a uma rotina menos intensa de estudos ou menor cobrança pessoal.

**Conclusão:** O desempenho acadêmico recompensa o equilíbrio. Sacrificar algumas horas de sono para estudar gera mais resultados do que dormir em excesso, mas a performance atinge seu limite máximo quando o aluno consegue manter 6 a 7 horas de descanso.

In [ ]:
grafico5_pct = df.groupby("screen_time_non_study")["cgpa_category"].value_counts(normalize=True).reset_index(name="Percentual")

df_grafico5 = px.bar(
    grafico5_pct,
    title="Tempo fora dos estudos",
    barmode="group",
    text_auto = ".1%",
    color="cgpa_category",
    x="screen_time_non_study",
    y="Percentual"

)

df_grafico5.show()

### O Impacto do Tempo de Tela Fora dos Estudos

Ao cruzar o tempo gasto em telas (lazer, redes sociais) com as notas finais, os dados quebram aquela regra de "quanto menos tela, melhor", revelando nuances sobre a qualidade do descanso:

* **O Excesso é Prejudicial:** Como esperado, o extremo superior (Mais de 6 horas) apresenta a menor taxa de notas de elite (13.9%), confirmando que o uso abusivo de telas concorre diretamente com o tempo e a qualidade do estudo.
* **O Paradoxo do Lazer Moderado:** Surpreendentemente, alunos que passam de 4 a 6 horas nas telas têm proporções de notas excelentes (16.7%) maiores do que aqueles que passam apenas 2 a 4 horas (16.1%).
* **O Grupo de Risco:** O dado mais curioso é que o grupo de "2 a 4 horas" concentra a maior taxa de alunos com notas baixas na faixa de 5.0 - 6.9 (24.5%). Isso sugere que não é apenas o *tempo* de tela que importa, mas como ele é usado: 4 a 6 horas podem representar um lazer estruturado (filmes, jogos) que alivia o estresse, enquanto 2 a 4 horas podem ser apenas distrações fragmentadas e ansiosas durante a rotina.

**Conclusão:** O lazer digital não é o grande vilão das notas, desde que não ultrapasse a marca de 6 horas. Um tempo de tela moderado (4-6 horas) parece fazer parte de uma rotina equilibrada e sustentável.

In [ ]:
grafico6_pct = df.groupby("study_consistency")["cgpa_category"].value_counts(normalize=True).reset_index(name="Percentual")

df_grafico6 = px.bar(
    grafico6_pct,
    title="Consistência",
    barmode="group",
    text_auto=".1%",
    color="cgpa_category",
    x="study_consistency",
    y="Percentual"
)

df_grafico6.show()

### O Poder da Consistência nos Estudos

Nossa última análise cruza a consistência da rotina de estudos com o desempenho final, provando que a disciplina diária supera o esforço esporádico:

* **O Prêmio da Constância:** O grupo "Mostly consistent" (Quase sempre consistente) é o líder absoluto no topo do boletim. 22.0% deles atingem a nota máxima (9.5 - 10.0), e formam o grupo com a menor taxa de notas baixas (apenas 10.9%).
* **O Custo da Inconstância:** Alunos que estudam de forma rara ("Rarely") sofrem o maior impacto negativo. Impressionantes 32.9% desse grupo tem as piores notas da turma (5.0 - 6.9), e eles praticamente não chegam ao topo (apenas 4.5% conseguem nota máxima).
* **O Meio-Termo:** O grupo "Sometimes" (Às vezes) fica exatamente no meio do caminho, mostrando um desempenho morno e distribuído.

**Conclusão:** A consistência é o pilar mais forte do desempenho acadêmico. O hábito regular de estudos protege o aluno da reprovação e é o caminho mais seguro para a excelência.

# Conclusão Final: O Raio-X do Desempenho Acadêmico Dos Alunos

Após conduzir a Análise Exploratória, conseguimos mapear de forma clara quais comportamentos impulsionam ou afundam as notas (CGPA) dos alunos.

A análise dos dados destruiu alguns mitos, como o de que o estágio atrapalha os estudos ou de que dormir muito é sempre melhor, e nos permitiu desenhar o perfil do aluno de alto desempenho:

1. **A Base do Sucesso (Tempo e Disciplina):** Não existe atalho. O grupo de elite é formado por alunos que estudam mais de 2 horas por dia (P1) e mantêm uma rotina de estudos consistente (P6). A inconstância é o fator que mais gera notas baixas.
2. **O Diferencial Prático:** Alunos que já estão no mercado estagiando ou desenvolvendo projetos ativamente não apenas conseguem equilibrar a rotina, como lideram o topo das notas máximas. A prática alimenta a teoria (P2).
3. **A Faca de Dois Gumes do Estresse:** O estresse máximo (Nível 5) surpreendentemente abriga a maior taxa de notas perfeitas, indicando que a alta cobrança gera resultados, mas possivelmente a um custo alto para a saúde mental. O perigo real está no estresse intermediário, que concentra as piores notas (P3).
4. **A Otimização do Descanso e Lazer:** O desempenho máximo exige otimização. Dormir 6 a 7 horas provou ser o "ponto de equilíbrio" perfeito (P4). Além disso, o lazer digital não é um vilão, desde que seja mantido na faixa de 4 a 6 horas; o uso excessivo (mais de 6 horas) é o que realmente destrói o rendimento (P5).

**Veredito Final:** O aluno com as maiores notas neste dataset não é um gênio recluso. Ele é um estudante disciplinado, que já coloca a mão na massa no mercado, dorme o suficiente para o cérebro funcionar (6-7h), tem seu momento de lazer controlado e usa o estresse da cobrança como combustível.

In [ ]:
!pip install sqlalchemy psycopg2-binary

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.3/4.3 MB 39.1 MB/s eta 0:00:00


In [ ]:
from sqlalchemy import create_engine

conexao_string ='postgresql://neondb_owner:npg_P0ShnNd2cvkE@ep-plain-truth-attfbac1-pooler.c-9.us-east-1.aws.neon.tech/neondb?sslmode=require&channel_binding=require'

entrega_neon = create_engine(conexao_string)

df.to_sql("estudantes_notas", con=entrega_neon, if_exists='replace', index=False)

print("Dados enviados com sucesso!")

Dados enviados com sucesso!
